app/services/tasks/download_stats_migration_service.rb

```
@@ -145,6 +175,32 @@ def fetch_local_cache_stats(after_timestamp, output_path)
       aggregated_data.values
     end
 
+    # Assuming the second line of the CSV file contains the start date, process it and return the date
+    # Example: # Start date: 20220101
+    def retrieve_start_date_from_csv(file_path)
+      File.open(file_path) do |file|
+        # Skip the first line
+        file.readline
+
+        # Read and process the second line
+        second_line = file.readline.strip
+        if second_line.start_with?('# Start date:')
+          date_str = second_line.split(':').last.strip
+          begin
+            start_date = DateTime.strptime(date_str, '%Y%m%d')
+            return start_date.strftime('%Y-%m-%d')  # Format to 'YYYY-MM-DD'
+          rescue ArgumentError
+            Rails.logger.error("Invalid date format '#{date_str}' in file #{file_path}")
+            return nil
+          end
+        else
+          Rails.logger.error("Error reading start date from #{file_path}. Second line does not contain a start date.")
+          Rails.logger.error("Found second line: #{second_line}")
+          return nil
```

Since the application can't really proceed without a start date, I'd suggest having both the cases where the start date can't be determined throw and error rather than return nil